hugr/src/algorithm/const_fold.rs

```diff
@@ -395,4 +462,44 @@ mod test {
         let expected = Value::false_val();
         assert_fully_folded(&h, &expected);
     }
+
```

Comment:
Is it worth also including the test from https://github.com/CQCL/hugr/issues/996 ?